src/django_components/component.py

```diff
@@ -2937,8 +3048,30 @@ def _render_impl(
         #       but instead can render one component at a time.
         #############################################################################
 
-        with _prepare_template(component, template_data) as template:
-            component_ctx.template_name = template.name
+        # TODO_v1 - Currently we have to pass `template_data` to `prepare_component_template()`,
```

Comment:
As the comment says, in v1 we will be able to simplify this logic once we drop support for `get_template_string()`, `get_template_name()`, and `get_template()`.